### ISSUE TITLE
Query regarding transcript in optimal transport

### ISSUE THREAD

**Issue by sauradip** [Pilhyeon/Learning-Action-Completeness-from-Points#6]
Thanks for making this awesome work publicly available !

I wanted to know what is the meaning if the term "transcript" in "search.py" ? I cannot understand the pattern why sometimes [0,1] is given and [1] is used sometimes. Can you kndly elaborate ?

**Comment by Pilhyeon** [comment]
Hello,
The variable "transcript" stores the action (background) sequence of the given video, where action and background instances correspond to 1 and 0, respectively.
For instance, "transcript=[0,1,0,1,0]" indicates that the video contains a total of two action instances and three background instances.
Also, it also provides the information that the video starts and ends with background instances.

To ease understanding, we elaborate on the logic in L70-84 as follows.
1) We first check if the first action seed appears at the first frame. If yes, we set the initial transcript to be [1], as the video starts with an action instance (L84).
2) Or, we next check whether there are any background seeds before the first action seed (L71). If yes, the initial transcript is set to [0, 1], as the video starts with a background instance (L79).
3) If no background seeds were found, the transcript will be initialized as [1], since the video starts with an action instance (L82).

In summary, we derive the "transcript" based on the seeds.

**Comment by sauradip** [comment]
Thank you for the excellent description.

Just additional query is what algorithm did you use for optimal transport. I observed something called "SingleTranscriptGrammar". Can you slightly elaborate or link to a paper which uses such transport algorithm?

**Comment by Pilhyeon** [comment]
Hi, I referenced a part of code in [NeuralNetwork-Viterbi](https://github.com/alexanderrichard/NeuralNetwork-Viterbi) for implementing the search process, which I have acknowledged in the readme.

You might grasp some help from it, although there are some differences.

**Comment by sauradip** [comment]
Thank you ! All my doubts have been clarified.

One additional question Is you have used points as GT action points as seeds , is it also possible to take GT action proposals as seeds ? Is seeds used as a means to find pseudo background ?

**Comment by Pilhyeon** [comment]
Yes, one may try using the full supervision (GT proposals) for generating seeds, though I haven't tried this.

In this case, unlike the point-level supervision, one can readily obtain the "real" background seeds instead of the pseudo ones.

**Comment by sauradip** [comment]
Thanks a lot for taking out time and answering my doubts.
